src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
+                }
 
-            std::map<std::string, JasmineGraphHashMapLocalStore> graphDBMapLocalStoresPgrnk;
-            if (JasmineGraphInstanceService::isGraphDBExists(graphID, partitionID)) {
-                JasmineGraphInstanceService::loadLocalStore(graphID, partitionID, graphDBMapLocalStoresPgrnk);
-            }
+                bzero((char *)&serv_addr, sizeof(serv_addr));
+                serv_addr.sin_family = AF_INET;
+                bcopy((char *)server->h_addr, (char *)&serv_addr.sin_addr.s_addr, server->h_length);
+                serv_addr.sin_port = htons(port);
+                if (Utils::connect_wrapper(sockfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
+                    instance_logger.log("ERROR connecting", "error");
+                    // TODO::exit
+                    return 0;
+                }
 
-            if (JasmineGraphInstanceService::isInstanceCentralStoreExists(graphID, partitionID)) {
-                JasmineGraphInstanceService::loadInstanceCentralStore(graphID, partitionID, graphDBMapCentralStores);
-            }
+                bzero(data, INSTANCE_DATA_LENGTH + 1);
+                int result_wr = write(sockfd, JasmineGraphInstanceProtocol::HANDSHAKE.c_str(),
+                                      JasmineGraphInstanceProtocol::HANDSHAKE.size());
 
-            graphDB = graphDBMapLocalStoresPgrnk[graphID + "_" + partitionID];
-            centralDB = graphDBMapCentralStores[graphID + "_centralstore_" + partitionID];
-
-            map<long, map<long, unordered_set<long>>> egonetMap =
-                calculateLocalEgoNet(graphID, partitionID, serverPort, graphDB, centralDB, workerSockets);
-
-            string instanceDataFolderLocation =
-                Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
-            string attributeFilePart = instanceDataFolderLocation + "/" + graphID + "_egonet_" + partitionID;
-            ofstream partfile;
-            partfile.open(attributeFilePart, std::fstream::trunc);
-            for (map<long, map<long, unordered_set<long>>>::iterator it = egonetMap.begin(); it != egonetMap.end();
-                 ++it) {
-                map<long, unordered_set<long>> egonetInternalMap = it->second;
-                for (map<long, unordered_set<long>>::iterator itm = egonetInternalMap.begin();
-                     itm != egonetInternalMap.end(); ++itm) {
-                    unordered_set<long> egonetInternalMapEdges = itm->second;
-                    for (unordered_set<long>::iterator ite = egonetInternalMapEdges.begin();
-                         ite != egonetInternalMapEdges.end(); ++ite) {
-                        partfile << to_string(it->first) << "\t" << to_string(itm->first) << "\t" << to_string(*ite)
-                                 << endl;
-                    }
+                if (result_wr < 0) {
+                    instance_logger.log("Error writing to socket", "error");
```

Comment:
Can we have a better error message here?

Reply:
It's now moved to the wrapper functions in `Utils`

Reply:
Ok